### ISSUE TITLE
Have you tried the smooth-ap loss on the VehicleID datasets

### ISSUE THREAD

**Issue by HouBiaoLiu** [JDAI-CV/fast-reid#334]
What about the results on VehicleID dataset if using the smooth-ap loss in your project

**Comment by L1aoXingyu** [comment]
I didn't try smoothAP in vehicleID, but the test protocol about vehicleID in smoothAP is different from ours.

**Comment by HouBiaoLiu** [comment]
how different? I find that Smooth-AP loss is stronger than triplet loss in the paper Smooth-AP: Smoothing the Path
Towards Large-Scale Image Retrieval. In this paper the test protocol is as followed. 
![image](https://user-images.githubusercontent.com/13248587/99045152-de1d8f00-25cb-11eb-94ff-18c2450dfe16.png)
![image](https://user-images.githubusercontent.com/13248587/99045239-fe4d4e00-25cb-11eb-948b-2afeae221760.png)

**Comment by HouBiaoLiu** [comment]
@L1aoXingyu Have you tried this:  All images are resized to 256*256. At training time, we use random crops and flips as augmentations, and at test time, a single center crop of size 224*224 is used.

**Comment by L1aoXingyu** [comment]
@HouBiaoLiu This is imagenet training tricks, you can try it. But in my opinion, there will be no difference.

**Comment by HouBiaoLiu** [comment]
smooth-ap论文给出的指标和cross-batch memory文章给出的指标都是基于测试集遍历抽取一张作为搜索集，其余作为测试集，其指标是可以复现的，我这边在vehicleid上训练，测试small时top1 95%, top5 97% top10 98%，与论文结果一致，如果用你这个工程采用一个类别随机抽一张作为搜索，其余作为测试集，在此测试协议下top1 仅为77% top5为90% top10为95%@LiaoXingyu

**Comment by HouBiaoLiu** [comment]
另外这个工程给出的在vehicleid上的结果，采用10-fold cross-validation，其top1 top5指标是在这个测试协议上给出的吗？最终指标是怎么给出的呢 与交叉验证的具体关系是什么

**Comment by JinkaiZheng** [comment]
您好！我们这里的十折交叉验证是按照vehicleID官方的测试协议来的。
具体的说，就是每个ID里面随机采样一张image作为gallery，其余的全部作为query，为了避免随机性，重复测试10次求平均。

**Comment by JinkaiZheng** [comment]
@HouBiaoLiu 您好！我们这里的十折交叉验证是按照vehicleID官方的测试协议来的。
具体的说，就是每个ID里面随机采样一张image作为gallery，其余的全部作为query，为了避免随机性，重复测试10次求平均。

**Comment by HouBiaoLiu** [comment]
嗯，请问这10次测试结果大吗，另外我注意到您最新的分布式运行环境代码,交叉熵损失是只计算单卡的，而triplet-loss计算过程中注释掉了收集不同卡上的特征数据，这个对结果影响大吗？ 另外请问下 这里的分布式环境貌似没有同步BN

**Comment by L1aoXingyu** [comment]
这是优化之后的多卡实现，速度更快，精度也不错，我在 person reid 上面试过了，vehicle reid 上还没有进行验证，最近比较忙，你有空也可以跑一下；
如果需要使用 `syncBN`，直接把配置文件里面的 `BN` 换车 `syncBN` 就可以了。

**Comment by HouBiaoLiu** [comment]
嗯 我测试了最新版的，IMS_PER_BATCH设为256，采用neck_feat:'after'，random erasing开启，4卡分布式训练，在vehicleid上测试结果，small medium large top1会比你贴出来的结果低2个点， top5结果差不多，请问这是什么原因导致呢@L1aoXingyu

**Comment by L1aoXingyu** [comment]
lr linear scale 了吗？另外 pk sampler 是怎么设置的？

**Comment by HouBiaoLiu** [comment]
配置文件 WARMUP_ITERS设置为10，我查看日志我发现迭代20次后就变为3.2e-4，而不是20轮，您代码里没有乘以相应的iters_per_epoch啊？pk sampler 默认设置的config/defaults.py中为C.DATALOADER.PK_SAMPLER = True  _C.DATALOADER.NAIVE_WAY = False

**Comment by L1aoXingyu** [comment]
wamrup_iter 设置1000，或者2000，现在这个按 iter 算，naive_way 设为 true

**Comment by HouBiaoLiu** [comment]
对于vehicleid 一个类别下不同车辆图像肯定来源于不同的相机编号，这样naive_way设为true或者false是一样的把@L1aoXingyu

**Comment by L1aoXingyu** [comment]
@HouBiaoLiu 你打开 naive_way 训练试试，是有差别的

**Comment by HouBiaoLiu** [comment]
loss配置为交叉熵，Contrastive Loss， 降维特征通道数设为512，base learning rate设置为7e-4，naive_way 设为 true, 4卡，batch-size设为256，可复现您的结果@L1aoXingyu

**Comment by L1aoXingyu** [comment]
> loss配置为交叉熵，Contrastive Loss， 降维特征通道数设为512，base learning rate设置为7e-4，naive_way 设为 true, 4卡，batch-size设为256，wamrup_iter 设置为3000，可复现您的结果@L1aoXingyu

你可以把 `config.yaml` 发出来大家参考一下 👍

**Comment by sky186** [comment]
@ HouBiaoLiu
您好，请问后续您还有做 smoothap相关的工作吗？ 是否实际训练上优于 triplet？

**Comment by TelBotDev** [comment]
> 降维特征通道数设为512

Does this mean FEAT_DIM in BACKBONE? It is amazing that same result is got after you have changed that from 2048 to 512.
